proto/marshal/marshal.py

```diff
@@ -122,6 +127,80 @@ def register_rule_class(rule_class: type):
 
         return register_rule_class
 
+    @staticmethod
+    def _throw_type_error(primitive):
+        raise TypeError(f"Unacceptable value of type {type(primitive)}")
+
+    @staticmethod
+    def _throw_value_error(proto_type, primitive):
+        raise ValueError(f"Unacceptable value {type(primitive)} for type {proto_type}")
+
+    def validate_primitives(self, field, primitive):
```

Comment:
This function is unfortunate, but is the only way to recreate immediate validation when attempting something like `my_obj.some_string_field = 123`.